lib/web/ui/usercontext.go

```diff
@@ -207,9 +207,22 @@ func NewUserContext(user types.User, userRoles services.RoleSet, features proto.
 	dbServerAccess := newAccess(userRoles, ctx, types.KindDatabaseServer)
 	dbAccess := newAccess(userRoles, ctx, types.KindDatabase)
 	kubeServerAccess := newAccess(userRoles, ctx, types.KindKubeServer)
-	requestAccess := newAccess(userRoles, ctx, types.KindAccessRequest)
 	desktopAccess := newAccess(userRoles, ctx, types.KindWindowsDesktop)
 	cnDiagnosticAccess := newAccess(userRoles, ctx, types.KindConnectionDiagnostic)
+	requestAccess := newAccess(userRoles, ctx, types.KindAccessRequest)
+
+	// If the user has a requestable role or any `search_as_roles`,
+	// they can create access requests for themselves even without an
+	// explicit `allow` rule
+	if len(userRoles.GetAllowedSearchAsRoles()) > 0 || len(accessCapabilities.RequestableRoles) > 0 {
+		requestAccess = access{
+			List:   true,
+			Read:   true,
+			Edit:   true,
```

Comment:
They should only be granted access to `list`, `read` and `create`. They shouldn't be able to have the ability to `edit` (approve/deny) and `remove` just as a result of having requestable roles defined

Reply:
What if you want to add a resource to the request before it gets approved? You'd need edit for that, right?

Reply:
Thanks, fixed in https://github.com/gravitational/teleport/pull/20209/commits/877f6127cdbe23d52bca52d8f712e6d96161946b

Also made the `access` role only have `list`, `read` and `create`.

Reply:
> What if you want to add a resource to the request before it gets approved? You'd need edit for that, right?

Yes. Also, you can't delete your own access request through the web UI if we set `Delete` to false here.

I think we can return all to true here for the web UI sake, and if they try to delete or edit an acess request that is not theirs, they will get a permission denied error